Create javadoc for only openj9 extensions

lumpfish opened the pull request:
Signed-off-by: Simon Rushton <rushton@uk.ibm.com>

pshipton commented:
The link to https://docs.oracle.com/javase/8/docs/api/com/sun/management/GarbageCollectorMXBean.html?is-external=true doesn't work.

I find it here:
https://docs.oracle.com/javase/8/docs/jre/api/management/extension/com/sun/management/GarbageCollectorMXBean.html

pshipton commented:
The same footer comments as Java 11 apply.
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/307#issuecomment-633647994
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/307#issuecomment-633648640

lumpfish commented:
> The link to https://docs.oracle.com/javase/8/docs/api/com/sun/management/GarbageCollectorMXBean.html?is-external=true doesn't work.

Well spotted.  Fixed now.

lumpfish commented:
> The same footer comments as Java 11 apply.
> [ibmruntimes/openj9-openjdk-jdk11#307 (comment)](https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/307#issuecomment-633647994)

Changed to match the (amended) jdk11 footer.

lumpfish commented on closed/make/Javadoc.gmk:
```diff
@@ -483,3 +479,268 @@ $(OPENJ9_JAVADOC_MARKER) : $(COREAPI_DOCSDIR)/index.html $(OPENJ9_MGMT_INDEXALL_
+OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE := $(OPENJ9_ONLY_SHAREDCLASSES_DOCDIR)/index.html
+OPENJ9_ONLY_SHAREDCLASSES_OPTIONS_FILE := $(DOCSTMPDIR)/openj9_only.sharedclasses.options
+
+openj9-only-sharedclassesdocs: $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```
Are you saying you'd like to see
```
openj9-only-sharedclassesdocs : $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```
rather than
```
openj9-only-sharedclassesdocs: $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```
?

keithc-ca replied:
Yes. Perhaps I'm being pedantic, but putting spaces on both sides of colons avoids the question of what this means:
```
C:/foo:bar
```
Is it
```
C : /foo:bar
```
or
```
C:/foo : bar
```
All those names are possible (even if unlikely, particularly on Windows). The separation makes it clear (most importantly to humans).

lumpfish replied:
OK, but the other targets in the file do not have a space either side of the colon.  I think they should be consistent within a makefile.  (I'm happy to make them all have a space either side of the colon, if that's the standard we're adopting).

keithc-ca replied:
Main.gmk had those spaces; I think we should aim to apply this rule to all makefiles within the 'closed' folder.

lumpfish replied:
I've update to include a space on either side of a target's colon.

lumpfish commented on closed/make/Javadoc.gmk:
```diff
@@ -483,3 +479,268 @@ $(OPENJ9_JAVADOC_MARKER) : $(COREAPI_DOCSDIR)/index.html $(OPENJ9_MGMT_INDEXALL_
+
+#############################################################################
+#
+# openj9 only-mgmtdocs
```
Fixed

keithc-ca commented on closed/make/Javadoc.gmk:
```diff
@@ -81,20 +81,20 @@ OPENJ9_SHAREDCLASSES_INDEX_FILE := $(OPENJ9_SHAREDCLASSES_DOCDIR)/index.html
+openj9-sharedclassesdocs : $(OPENJ9_SHAREDCLASSES_INDEX_FILE)
 
 # Set relative location to platform document root
 $(OPENJ9_SHAREDCLASSES_INDEX_FILE): GET2DOCSDIR := $(PLATFORM_DOCSDIR)/..
```
Space around `:` - here, and lines 280 & 560. Also, the additions to `closed/make/Main.gmk`.

lumpfish replied:
Fixed

keithc-ca commented on closed/make/Javadoc.gmk:
```diff
@@ -371,8 +367,8 @@ OPENJ9_MGMT_SRCPATH := $(OPENJ9_DOCS_SRCDIR)$(PATH_SEP)$(TEMP_SUN_MGMT_PKGDIR)
+$(OPENJ9_MGMT_INDEXALL_FILE) : FIRST_COPYRIGHT_YEAR := $(OPENJ9_MGMT_FIRST_COPYRIGHT_YEAR)
+$(OPENJ9_MGMT_INDEXALL_FILE) : $(OPENJ9_MGMT_INDEX_FILE) $(OPENJ9_MGMT_OPTIONS_FILE) $(OPENJ9_MGMT_PACKAGES_FILE)
 	$(MKDIR) -p $(TEMP_SUN_MGMT_SRCDIR)
 	if [ ! -d $(TEMP_SUN_MGMT_SRCDIR) ]; then echo Error: $(TEMP_SUN_MGMT_SRCDIR) does not exist, mkdir -p $(TEMP_SUN_MGMT_SRCDIR) must have failed; exit 1; fi
```
What is the point of this command?
If it needs to stay, it should use `$(ECHO)` and `$(MKDIR)` and the message should be quoted.

lumpfish replied:
I've removed it.  If the` mkdir -p`  fails we would get an error message from that.

keithc-ca commented:
This looks good. It's time to squash and when the equivalent changes are ready for all versions (see https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/307#issuecomment-636977105) I'm prepared to merge.

lumpfish commented:
I've updated the PR with these changes following a review by the openj9 documentation team:
1. Added the openj9 dtfj and zosconditionhandling packages
2. Made external links open in a new tab
3. Fixed a bug where changing the value of JAVASE_BASE_URL part way through the makefile meant that a target could have the incorrect value when it ran and create invalid href links.  Changed to use a unique variable for each target.

keithc-ca commented:
If you anticipate no further functional changes, it's time to squash in preparation for merging.

lumpfish commented:
Squashed
